pkg/cmd/adm/unregister_member_test.go

```diff
@@ -29,16 +29,17 @@ func TestUnregisterMemberWhenAnswerIsY(t *testing.T) {
 	err := UnregisterMemberCluster(ctx, "member1")
 
 	// then
-	require.NoError(t, err)
+	require.ErrorContains(t, err, "no such host") // since we have not set up http client required for restart(),it will throw an error.
```

Comment:
Mocking a restartFunc here requires a lot more complex changes , hence to avoid a chain of changes and to make the test case simple, i have tested it for a specific error that only comes when it tries to implement rollout status from kubectl.

Reply:
let's not expect an error when there shouldn't be any, this is not a good practice.

if you modify the `UnregisterMemberCluster` function to:
```go
func UnregisterMemberCluster(ctx *clicontext.CommandContext, clusterName string, restartDeployments func(ctx *clicontext.CommandContext, clusterNames ...string) error) error {
```
calling it:
```go
return UnregisterMemberCluster(ctx, args[0], restart)
```
so you can use the parameter:
```go
return restartDeployments(ctx, "host")
```
and in the tests, you can create something like:
```go
func restartDeploymentInterceptor() (func(ctx *clicontext.CommandContext, clusterNames ...string) error, map[string]int) {
	interceptedCalls := map[string]int{}
	return func(ctx *clicontext.CommandContext, clusterNames ...string) error {
		for _, clusterName := range clusterNames {
			interceptedCalls[clusterName]++
		}
		return nil
	}, interceptedCalls
}
```
and use it:
```go
restartFunc, interceptor := restartDeploymentInterceptor()
// when
err := UnregisterMemberCluster(ctx, "member1", restartFunc)
// then
require.NoError(t, err)
assert.Len(t, interceptor, 1)
assert.Equal(t, 1, interceptor["host"])
```

But you can play with the `restartDeploymentInterceptor` function - it can return an error to verify an error case, etc... (if needed)

Reply:
> i have changed it now . Please ignore the previous comment

I've started responding before you pushed the latest changes, so I didn't see your comment nor the last commit